elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java

```diff
@@ -954,34 +956,40 @@ protected QueryBuilder getFilterForContainsPredicate(Contains contains, HasValue
 
         List<QueryBuilder> filters = new ArrayList<>();
         for (String propertyName : propertyNames) {
-            if (Element.ID_PROPERTY_NAME.equals(propertyName)) {
-                propertyName = Elasticsearch5SearchIndex.ELEMENT_ID_FIELD_NAME;
-            } else if (Edge.LABEL_PROPERTY_NAME.equals(propertyName)) {
-                propertyName = Elasticsearch5SearchIndex.EDGE_LABEL_FIELD_NAME;
-            } else if (Edge.OUT_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
-                propertyName = Elasticsearch5SearchIndex.OUT_VERTEX_ID_FIELD_NAME;
-            } else if (Edge.IN_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
-                propertyName = Elasticsearch5SearchIndex.IN_VERTEX_ID_FIELD_NAME;
-            } else if (value instanceof String
-                    || value instanceof String[]
-                    || (value instanceof Object[] && ((Object[]) value).length > 0 && ((Object[]) value)[0] instanceof String)
-                    ) {
-                propertyName = propertyName + Elasticsearch5SearchIndex.EXACT_MATCH_PROPERTY_NAME_SUFFIX;
-            }
-            switch (contains) {
-                case IN:
-                    filters.add(QueryBuilders.termsQuery(propertyName, (Object[]) value));
-                    break;
-                case NOT_IN:
-                    filters.add(QueryBuilders.boolQuery().mustNot(QueryBuilders.termsQuery(propertyName, (Object[]) value)));
-                    break;
-                default:
-                    throw new VertexiumException("Unexpected Contains predicate " + has.predicate);
-            }
+            filters.add(getFilterForProperty(contains, has, propertyName, value));
         }
         return getSingleFilterOrOrTheFilters(filters, has);
     }
 
+    private QueryBuilder getFilterForProperty(Contains contains, HasValueContainer has, String propertyName, Object value) {
+        if (Element.ID_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.ELEMENT_ID_FIELD_NAME;
+        } else if (Edge.LABEL_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.EDGE_LABEL_FIELD_NAME;
+        } else if (Edge.OUT_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.OUT_VERTEX_ID_FIELD_NAME;
+        } else if (Edge.IN_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.IN_VERTEX_ID_FIELD_NAME;
+        } else if (Edge.IN_OR_OUT_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
+            return QueryBuilders.boolQuery()
+                    .should(getFilterForProperty(contains, has, Edge.OUT_VERTEX_ID_PROPERTY_NAME, value))
+                    .should(getFilterForProperty(contains, has, Edge.IN_VERTEX_ID_PROPERTY_NAME, value));
```

Comment:
Since this is an `or` with possibly other criteria involved, should we be safe and set the `minimum_should_match` to `1`? This is from the docs for `should`:
```
If the bool query is in a query context and has a must or 
filter clause then a document will match the bool query even
if none of the should queries match.  In this case these 
clauses are only used to influence the score.
...
This behavior may be explicitly controlled by settings the minimum_should_match parameter.
```